lib/src/widgets/reaction_popup.dart

```diff
@@ -147,11 +147,13 @@ class ReactionPopupState extends State<ReactionPopup>
         onEmojiTap: (emoji) {
           widget.onTap();
           if (currentUser != null && _message != null) {
-            chatController?.setReaction(
-              emoji: emoji,
-              messageId: _message!.id,
-              userId: currentUser!.id,
-            );
+            if (!(reactionPopupConfig?.overrideUserReactionCallback ?? false)) {
```

Comment:
```suggestion
            if (!(reactionPopupConfig?.overrideUserReactionCallback ?? true))
```
I think we need to use `true` as the default because we are inverting the condition. If `overrideUserReactionCallback` is null, the default should be `false`, which will correctly invert to `true`. That's why we should use `true` as the default here.

Reply:
You are missing brackets here. if it's null it should be the default value which is false in constructor.

Reply:
For example, if the user has not set up reactionPopupConfig, it will be null. As a result, this condition will evaluate as true, allowing the system to function as it currently does for existing users.

Reply:
Got it.